### PR TITLE
Refactor: VoteHandler.accept_vote returns Option 

### PR DESCRIPTION
**Pull request by tisonkun**
**Checklist**

- [ ] Updated guide with pertinent info (may not always apply). 
- [x] Squash down commits to one or two logical commits which clearly describe the work you've done.
- [x] Unittest is a friend:)

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/openraft/863)

### CODE REVIEW

**Review by drmingdrmer — COMMENTED** [review]
LGTM. Thank you~

File: openraft/src/engine/handler/vote_handler/mod.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ where C: RaftTypeConfig
         vote: &Vote<C::NodeId>,
         tx: ResultSender<T, E>,
         f: F,
-    ) -> Result<ResultSender<T, E>, ()>
+    ) -> Option<ResultSender<T, E>>
```
[the remark this answers is not in the document]
Comment (drmingdrmer) [reply openraft/src/engine/handler/vote_handler/mod.rs:55]: @tisonkun 
The doc comment for this method needs to be updated.

[the remark this answers is not in the document]
Comment (tisonkun) [reply openraft/src/engine/handler/vote_handler/mod.rs:55]: OK. Let me handle it today.

[the remark this answers is not in the document]
Comment (tisonkun) [reply openraft/src/engine/handler/vote_handler/mod.rs:55]: Updated. PTAL @drmingdrmer.

**Review by drmingdrmer — APPROVED** [review]

File: openraft/src/engine/handler/vote_handler/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (drmingdrmer) [reply openraft/src/engine/handler/vote_handler/mod.rs:55]: You may need to squash commits into one:)

**Comment by drmingdrmer** [comment]
@mergify update

**Comment by drmingdrmer** [comment]
Manually merged in: 5593b96dea30979a1bdc924abbfe3fc8d2463ecd

File: openraft/src/engine/handler/vote_handler/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tisonkun) [reply openraft/src/engine/handler/vote_handler/mod.rs]: Aha. That's megirfy's restriction. Generally I use GitHub's Squash and merge and it doesn't matter how the contributor organize their commits.

**Pull request closed without merging**